### ISSUE TITLE
getUpdateHandler returns incorrect errors from validation

### ISSUE THREAD

**Issue by ddaddy** [keystonejs/keystone#2923]
When creating an object and saving it, it runs through validation and returns helpful errors indicating the `path` at the field that failed validation.
However when using getUpdateHandler it returns errors with the path undefined.

I have a simple User object:
```
User.add({
	name: { type: Types.Name, required: true, index: true },
	email: { type: Types.Email, initial: true, required: true, index: true },
	password: { type: Types.Password, initial: true, required: true },
	resetPasswordKey: { type: String, hidden: true }
}, 'Permissions', {
	isAdmin: { type: Boolean, label: 'Can access Keystone', index: true }
});
```
and on a form post I update it like this:
```
view.on('post', {action: 'save'}, function(next) 
{
	var user = req.user;
	user.getUpdateHandler(req).process(req.body, {
		fields: 'name, email',
		flashErrors: true
	}, function(err) {
	
		if (err) {
			console.log(err);
			next();
		} else {
			req.flash('success', { detail: 'Your profile has been saved'});
			res.redirect('/index');
		}
	});
});
```
My form only contains 3 fields. `Name.first`, `Name.last` & `email`.

If I make both name fields empty and submit, I get a nice warning message that `Name is required`, however if I leave email blank I get an empty red box and the error object is undefined:
```
{ message: 'Validation failed',
  name: 'ValidatorError',
  errors: 
   { undefined: 
      { name: 'ValidatorError',
        path: undefined,
        message: 'Validation failed',
        type: 'required' } } }
```

Additionally to this, I experienced similar problems in another object when I changed my method from save to getUpdateHandler as it was the only way I could get `CloudinaryImage` to automatically upload, and in this case I had custom validation like this:
```
Property.schema.path('content.brief').validate(function (brief) 
{
	if (!brief) return true;
	return brief && brief.length < 100;
}, 'Your brief description cannot be more than 100 characters');
```
yet if the validation failed it returned an undefined error like above.

Additionally, it seemed that it would only first complain about validation for required fields, then once they were filled in it would complain about the additional validation.

**Comment by geloescht** [comment]
Is this on v0.3 or the current master (v0.4 alpha)?

**Comment by ddaddy** [comment]
I believe it's v0.3 from NPM

**Comment by r3wt** [comment]
i can confirm this. google brought me here

**Comment by r3wt** [comment]
this is so stupid.

**Comment by r3wt** [comment]
I had the same or atleast a similar problem, but found a workaround.

```js
	view.on('post', { action: 'update' }, function(next) {
		
		var editUser = req.user,
			updater = editUser.getUpdateHandler(req);
			
		var fields = 'name,email,image,bio,title,contact.email,contact.phone';
		
		var opt = ['password','role'];
		
		for(var i=0;i<opt.length;i++){
			if(req.body.hasOwnProperty(opt[i])){
				if( req.body[opt[i]].length == 0){
					continue;
				}
				fields +=','+opt[i];
			}
		}
		
		updater.process( req.body, {
			flashErrors: true,
			fields: fields,
			errorMessage: 'There was a problem saving your profile: '
		}, function(err) {
			console.log(err);
			if (err) {
				locals.validationErrors = err.errors;
				locals.profileSaved = false;
			}else{
				locals.profileSaved = true;
			}
			next();
		});
		
	});
```
it seems that the failure was related to `role` and `password` being required fields. when i place them into the `opt` if they are not in the body they do not get processed. at the same time, i also had an non existent field in the fields parameter

**Comment by mxstbr** [comment]
This should be fixed with the complete UpdateHandler rewrite that was merged in #3319. I'll close this for now, if it turns out to still be an issue for you with those updates please comment again!
